scripts/docker/build_and_test.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
+
+build_dir="/workspace/build"
+
+# ========================
+# clean
+# ========================
+if [[ "${CLEAN_BUILD}" = "true" && -d ${build_dir} ]]; then
+  print_text_box "Clean build"
+  rm -fr ${build_dir}
+  echo "Removed ${build_dir}"
+fi
+
+# ========================
+# configure
+# ========================
+print_text_box "Configure build"
+cmake \
+  -S . \
+  -B ${build_dir} \
+  -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
```

Comment:
A potential enhancement would be to verify that all specified environment variables are set; if any are missing, an error should be raised.

Reply:
what can be check is the validity value of the strings for ex build type is one of "release", "debug" or "relwithdebinfo". simulated booleans can be "true" or "false". Will do it in separate issue.